src/VisualStudio/Core/Def/Implementation/Workspace/Esent/EsentPersistentStorage.cs

```diff
@@ -274,12 +274,10 @@ private bool TryGetUniqueId(string value, bool fileCheck, out int id)
 
             try
             {
-                id = _nameTableCache.GetOrAdd(value, v =>
-                {
-                    // okay, get one from esent
-                    var uniqueIdValue = fileCheck ? FilePathUtilities.GetRelativePath(Path.GetDirectoryName(SolutionFilePath), v) : v;
-                    return _esentStorage.GetUniqueId(uniqueIdValue);
-                });
+                var uniqueIdValue = fileCheck ? FilePathUtilities.GetRelativePath(Path.GetDirectoryName(SolutionFilePath), value) : value;
+                id = _nameTableCache.GetOrAdd(value, _esentStorage.GetUniqueId(uniqueIdValue));
```

Comment:
So, say the value is already in the name table.  This will still cause us to get a unique id from esent storage, which we will then not use.  Is that ok?  Will this cause us to somehow run out of these unique IDs?  Does this constitute a leak?  I dont' know enough about the esent layer, so i need more info to know if this is ok.

I feel like it would be safer to just have two cached delegates that we use instead.